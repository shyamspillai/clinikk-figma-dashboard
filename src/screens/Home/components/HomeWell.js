import styled from 'styled-components'

const HomeWell = () => {
    return (
        <HomeWellComponent>
            <span className="baloo weight-600 size-60 font-black">Hello Josh!</span>
            <span className="roboto size-16 weight-400 font-black">It's good to see you again</span>
        </HomeWellComponent>
    )
}

const HomeWellComponent = styled.div`
    flex: 0.5;
    height: 160px;
    align-items: flex-start;
    justify-content: center;
    padding: 10px 20px;
    background-color: #f5f5f7;
    border-radius: 15px;
    display: flex;
    flex-direction: column;

    @media only screen and (max-width: 400px) {
        width: 300px !important;
        display: flex;
        margin-bottom: 30px;
    }
`

export default HomeWell
