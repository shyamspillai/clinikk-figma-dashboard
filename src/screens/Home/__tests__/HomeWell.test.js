import React from 'react';
import HomeWell from '../components/HomeWell'

import Enzyme, { render, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

describe('Home Well', () => {

  it('should match snapshot', () => {
    const HomeWellComponent = shallow(<HomeWell />);
    expect(toJson(HomeWellComponent)).toMatchSnapshot();
  })

  it('renders the main title', () => {
    const wrapper = render(
      <HomeWell />
    );
    const element = wrapper.find('span.baloo').first();
    expect(element.text()).toEqual('Hello Josh!');

  });

  it('renders the subtitle messaging', () => {
    const wrapper = render(
      <HomeWell />
    );
    const element = wrapper.find('span.roboto').first()
    expect(element.text()).toEqual("It's good to see you again");

  });

  it('should render the main title even on resize', () => {

    const wrapper = render(
      <HomeWell />
    );

    // Change the viewport to 500px.
    global.innerWidth = 500;

    // Trigger the window resize event.
    global.dispatchEvent(new Event('resize'));

    const element = wrapper.find('span.baloo').first();
    expect(element.text()).toEqual('Hello Josh!');
  })

  it('should render the subtitle even on resize', () => {

    const wrapper = render(
      <HomeWell />
    );

    // Change the viewport to 500px.
    global.innerWidth = 500;

    // Trigger the window resize event.
    global.dispatchEvent(new Event('resize'));

    const element = wrapper.find('span.roboto').first()
    expect(element.text()).toEqual("It's good to see you again");
  })

});