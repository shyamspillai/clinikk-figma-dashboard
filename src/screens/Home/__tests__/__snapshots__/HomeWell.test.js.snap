// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Well should match snapshot 1`] = `
<styled.div>
  <span
    className="baloo weight-600 size-60 font-black"
  >
    Hello Josh!
  </span>
  <span
    className="roboto size-16 weight-400 font-black"
  >
    It's good to see you again
  </span>
</styled.div>
`;
